### PR TITLE
update what4, crucible, and what4-serialize to master

### PR DESCRIPTION
**Pull request by benjaminselfridge**
This brings us up to date on asl-translator w.r.t. the bv-sized changes.

### CODE REVIEW

File: asl-translator.cabal [hunk 1]
```diff
@@ -22,7 +22,7 @@ flag unsafe-operations
 
 flag asl-lite
   Description: Use a trimmed-down set of instructions for the ASL specification (sufficient to disassemble u-boot).
-  Default: False
+  Default: True
```
Comment (travitch) [inline asl-translator.cabal:25]: This should be a separate commit if we do it - it would require a coordinating change with dismantle

File: asl-translator.cabal [hunk 2]
```diff
@@ -133,6 +133,7 @@ test-suite asl-translator-formula-test
 executable asl-translator-concrete-test
   main-is: Concrete.hs
   build-depends:       base >= 4.10.0.0 && < 5,
+                       bv-sized >= 1.0.0,
```
Comment (travitch) [inline asl-translator.cabal:136]: My usual upper bound comment

File: asl-translator.cabal (hunk 1 above)
Comment (benjaminselfridge) [reply asl-translator.cabal:25]: Oops -- I think that is a mistake.

File: asl-translator.cabal (hunk 2 above)
Comment (benjaminselfridge) [reply asl-translator.cabal:136]: Actually I think maybe this shouldn't have any bound at all, so it implicitly will use the library version?

Comment (travitch) [reply asl-translator.cabal:136]: Oh, you are right - I was overzealous and didn't realize this was a test stanza

Comment (benjaminselfridge) [reply asl-translator.cabal]: I've removed the bound

File: asl-translator.cabal (hunk 1 above)
Comment (benjaminselfridge) [reply asl-translator.cabal]: I've changed it back to False
